cmd/tools/gen-accessors.go

```diff
@@ -0,0 +1,344 @@
+//+build ignore
+
+/*
+ * Datadog API for Go
+ *
+ * Please see the included LICENSE file for licensing information.
+ *
+ * Copyright 2017 by authors and contributors.
+
+ gen-accessors generates GetXx, GetXxOk, HasXx and SetXx methods for struct field types with pointers.
+
+ It should be run by the go-datadog-api authors to ge-generate methods before pushing changes to GitHub.
+*/
+
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"go/ast"
+	"go/format"
+	"go/parser"
+	"go/token"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+	"text/template"
+	"time"
+)
+
+const (
+	fileSuffix = "-accessors.go"
+)
+
+var (
+	verbose    = flag.Bool("v", false, "Print verbose log messages")
+	sourceTmpl = template.Must(template.New("source").Parse(source))
+)
+
+func logf(fmt string, args ...interface{}) {
+	if *verbose {
+		log.Printf(fmt, args...)
+	}
+}
+
+func main() {
+	flag.Parse()
+	fset := token.NewFileSet()
+
+	pkgs, err := parser.ParseDir(fset, ".", sourceFilter, 0)
+	if err != nil {
+		log.Fatal(err)
+		return
+	}
+
+	for pkgName, pkg := range pkgs {
+		t := &templateData{
+			Year:     time.Now().Year(),
+			Filename: pkgName + fileSuffix,
+			Package:  pkgName,
+			Imports:  map[string]string{},
+		}
+		for filename, f := range pkg.Files {
+			logf("Processing %v...", filename)
+			t.sourceFile = filename
+			if err := t.processAST(f); err != nil {
+				log.Fatal(err)
+			}
+		}
+		if err := t.dump(); err != nil {
+			log.Fatal(err)
+		}
+	}
+	logf("Done.")
+}
+
+// processAST walks source code files, looks for struct definitions, and calls appropriate helper functions
+// to add generate accessors for type fields are star expressions -ie: pointers
+func (t *templateData) processAST(f *ast.File) error {
```

Comment:
This currently only goes one level deep. If we want to support nested structs this will needs refactoring. Although we have nested structs at the moment, I'm not super keen on them.

My preference at the moment is to get all this merged into v2, and if we agree on the direction, support accessors for nested structs in a separate PR. It's not much work but this PR is a big of a change as it is.

AFAIC, before we make v2 the default branch we can safely to this and test with v2.

@zorkian thoughts?